### PR TITLE
User adds word

### PR DESCRIPTION
**Pull request by MarcvanMelle**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -61,11 +61,15 @@ def admin_redirect
   end
 
   def user_valid_destroy
-    @user.reviews.delete_all
+    @user.reviews.each do |review|
```
[the remark this answers is not in the document]
Comment (alxjrvs) [reply app/controllers/users_controller.rb]: Hound, you complicated. 

Hound is suggesting you do this: 

`@user.reviews.each(&:delete)`.

[the remark this answers is not in the document]
Comment (alxjrvs) [reply app/controllers/users_controller.rb]: I don't think you really have to, but what's going on is actually really interesting: 

You are defining (through very non-obvious symbols) that you are creating a block-symbol: `&:delete`. This represents a block that then sends a message to the method that called it. 

You then pass that block to `each` (same as you do when using `do...end` blocks) which sends that message _in turn_ to each item that you are iterating over. 

So, this: 

`@user.reviews.each(&:delete)`

is functionally the same as: 

```
@user.reviews.each do |r| 
  r.delete
end
```

[the remark this answers is not in the document]
Comment (alxjrvs) [reply app/controllers/users_controller.rb]: I think you can leave it as is and silence the Hound. 

ALSO - last nitpick - favor `destroy` over `delete`. `delete` just executes a delete statement in SQL, whereas `destroy` uses the ActiveRecord Lifecycle.

File: app/controllers/words_controller.rb [hunk 2]
```diff
@@ -14,6 +14,28 @@ def show
...
+    if current_user
+      @word = Word.new
+    else
+      flash[:errors] = "You must be logged in to create a word."
+      redirect_to root_path
+    end
+  end
+
+  def create
+    if current_user
```
Comment (alxjrvs) [inline app/controllers/words_controller.rb:27]: If you have a top-level `if` statement with no `else` in a controller action, what you actually want is a Controller Filter. Do some googling and see if you can suss it out!

Comment (MarcvanMelle) [reply app/controllers/words_controller.rb:27]: `@word = Word.new(...) if current_user` yeah?

**Pull request merged**